Main.cc

```diff
@@ -136,15 +137,26 @@ void OnMessage(uWS::WebSocket<uWS::SERVER>* ws, char* raw_message, size_t length
                     if (message.histograms_size() == 0) {
                         session->CancelSetHistRequirements();
                     } else {
-                        tsk = new (tbb::task::allocate_root(session->context()))
+                        tsk = new (tbb::task::allocate_root(session->HistoContext()))
```

Comment:
It seems have to reset the `_histo_context` like adding `session->HistoContext().reset();` before this line. Otherwise, the new Per-Cube process cannot be started once have set `_histo_context.cancel_group_execution()`.